src/librustc_metadata/dynamic_lib.rs

```diff
@@ -77,16 +77,35 @@ mod dl {
             // dlerror isn't thread safe, so we need to lock around this entire
             // sequence
             let _guard = (*LOCK).lock();
+
+            // dlerror reports the most recent failure that occured during a
+            // dynamic linking operation and then clears that error; we call
+            // once in advance of our operation in an attempt to discard any
+            // stale prior error report that may exist:
             let _old_error = libc::dlerror();
```

Comment:
Is this call still needed? Surely any prior error will be replaced if there's a new error.

Reply:
As @ollie27 said, there's no need to do this anymore if we don't use the return value of `dlerror` to determine whether an error occurred.